v3/docs/07-advanced/h-ss58/index.mdx

```diff
@@ -7,28 +7,128 @@ category: advanced
+- 10000000b..=11111111b (128..=255 inclusive) Reserved for future address format extensions.
+
+The latter (42) is a "wildcard" address that is meant to be equally valid on all Substrate networks that support fixed-length addresses.
+For production networks, however, a network-specific version may be desirable to help avoid the key-reuse between networks and some of the problems that it can cause.
+Substrate Node will default to printing keys in address type 42,
+though alternative Substrate-based node implementations (e.g. Polkadot) may elect to default to some other type.
+
+### Address Formats for Substrate
+
+There are 16 different address formats, identified by the length (**in bytes**) of the total payload (i.e. including the checksum).
+
+| Total | Type | Raw Account | Checksum |
+|:-----:|:----:|:-----------:|:--------:|
+| 3     | 1    | 1           | 1        |
+| 4     | 1    | 2           | 1        |
+| 5     | 1    | 2           | 2        |
+| 6     | 1    | 4           | 1        |
+| 7     | 1    | 4           | 2        |
+| 8     | 1    | 4           | 3        |
+| 9     | 1    | 4           | 4        |
+| 10    | 1    | 8           | 1        |
+| 11    | 1    | 8           | 2        |
+| 12    | 1    | 8           | 3        |
+| 13    | 1    | 8           | 4        |
+| 14    | 1    | 8           | 5        |
+| 15    | 1    | 8           | 6        |
+| 16    | 1    | 8           | 7        |
+| 17    | 1    | 8           | 8        |
+| 34    | 1    | 32          | 2        |
+
+## Checksum types
+
+Several potential checksum strategies exist within Substrate, giving different length and longevity guarantees.
+There are two types of checksum preimage (known as SS58 and AccountID) and many different checksum lengths (1 to 8 bytes).
+
+In all cases for Substrate, the [Blake2-256](https://en.wikipedia.org/wiki/BLAKE_(hash_function)) hash function is used.
+The variants simply select the preimage used as the input to the hash function and the number of bytes taken from its output.
+
+The bytes used are always the left most bytes.
+The input to be used is the non-checksum portion of the SS58 byte series used as input to the base-58 function, i.e. `concat( <address-type>, <address> )`.
+A context prefix of `0x53533538505245`, (the string `SS58PRE`) is prepended to the input to give the final hashing preimage.
+
+The advantage of using more checksum bytes is simply that more bytes provide a greater degree of protection against input errors and
+index alteration at the cost of widening the textual address by an extra few characters.
+For the account ID form, this is insignificant and therefore no 1-byte alternative is provided.
+For the shorter account-index formats, the extra byte represents a far greater portion of the final address and
+so it is left for further up the stack (though not necessarily the user themself) to determine the best tradeoff for their purposes.
```

Comment:
```suggestion
The advantage of using more checksum bytes is simply that more bytes provide a greater degree of protection against input errors and index alteration at the cost of widening the textual address by an extra few characters.
For the account ID format, this is insignificant and therefore no 1-byte alternative is provided.
For the shorter account-index formats, the extra byte represents a far greater portion of the final address, so it is left for further up the stack (though not necessarily the user themself) to determine the best tradeoff for their purposes.
```
The last sentence isn't really clear to me especially:
> left for further up the stack 
> though not necessarily the user themself

@NukeManDan can you try simplifying here? 🙏🏻